In [94]:
import os
from pathlib import Path
import pandas as pd


CURRENT_DIR = os.path.abspath('')
DATA_DIR = Path(CURRENT_DIR).parent / "data"





In [95]:
FILE_PATH = DATA_DIR / "sentiment_values.csv"

df = pd.read_csv(FILE_PATH)
df = df.rename(columns = {'hashtag' : 'ss_score'})
df = df.reset_index()
df = df.rename(columns = {'level_0':'hashtag','level_1':'vader_score','level_2':'afinn_score','level_3':'ol_score'})

In [96]:
#Set columns
df.columns=['hashtag','vader_score','afinn_score','ol_score','ss_score','vader_min','vader_max','vader_sum','vader_avg','afinn_min',
            'afinn_max','afinn_sum','afinn_avg','ol_min','ol_max','ol_sum','ol_avg','ss_min','ss_max','ss_sum','ss_avg']

df = df.drop(['vader_min','vader_max','vader_sum','afinn_min','afinn_max','afinn_sum','ol_min','ol_max','ol_sum','ss_min','ss_max','ss_sum'], axis=1)
df.head()

hashtag  vader_score  afinn_score  ol_score  ss_score  vader_avg   
0          relaxtime          0.8          0.8       2.4       0.8        NaN  \
1  melovechilicheese          0.8          0.8       0.8       0.8        NaN   
2         greatmusic          0.8          0.8       2.4       0.8        1.0   
3         rockballad          0.7          0.7       0.7       0.7        NaN   
4         amonamarth          0.3          0.3       0.3       0.3        NaN   

   afinn_avg  ol_avg  ss_avg  
0     0.7375     NaN     NaN  
1     0.9000     1.0     0.8  
2     0.8875     1.0     0.8  
3        NaN     NaN     NaN  
4        NaN     0.0     NaN

In [97]:
len(df['hashtag'].unique().tolist())
len(df['hashtag'].unique().tolist())
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5290 entries, 0 to 5289
Data columns (total 9 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   hashtag      5290 non-null   object 
 1   vader_score  3867 non-null   float64
 2   afinn_score  3867 non-null   float64
 3   ol_score     3867 non-null   float64
 4   ss_score     3867 non-null   float64
 5   vader_avg    255 non-null    float64
 6   afinn_avg    2635 non-null   float64
 7   ol_avg       2823 non-null   float64
 8   ss_avg       2160 non-null   float64
dtypes: float64(8), object(1)
memory usage: 372.1+ KB


In [98]:
import numpy as np
df['vader_score'] = df.apply(lambda row: row['vader_avg'] if np.isnan(row['vader_score']) else row[['vader_score']], axis=1)
df['afinn_score'] = df.apply(lambda row: row['afinn_avg'] if np.isnan(row['afinn_score']) else row['afinn_score'], axis=1)
df['ol_score'] = df.apply(lambda row: row['ol_avg'] if np.isnan(row['ol_score']) else row['ol_score'], axis=1)
df['ss_score'] = df.apply(lambda row: row['ss_avg'] if np.isnan(row['ss_score']) else row['ss_score'], axis=1)


In [99]:
df.info()
#vader_score didn't change
#afinn_score increased from 3867 to 4532
#ol_score increased from 3867 to 4831
#ss_score increased from 3867 to 4471

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5290 entries, 0 to 5289
Data columns (total 9 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   hashtag      5290 non-null   object 
 1   vader_score  3871 non-null   float64
 2   afinn_score  4532 non-null   float64
 3   ol_score     4831 non-null   float64
 4   ss_score     4471 non-null   float64
 5   vader_avg    255 non-null    float64
 6   afinn_avg    2635 non-null   float64
 7   ol_avg       2823 non-null   float64
 8   ss_avg       2160 non-null   float64
dtypes: float64(8), object(1)
memory usage: 372.1+ KB


In [100]:
# #Remove all of the unnecessary scores - ol_score has the highest amount of ratings per hashtag # option1
#df1 = df.drop(['vader_score','afinn_score','ss_score','vader_avg','afinn_avg','ol_avg','ss_avg'], axis=1)

# or remove only "meta" scores used to compute each score but keep the four distinct scores 
df1 = df.drop(['vader_avg','afinn_avg','ol_avg','ss_avg'], axis=1)
df1.head(5)

hashtag  vader_score  afinn_score  ol_score  ss_score
0          relaxtime          0.8          0.8       2.4       0.8
1  melovechilicheese          0.8          0.8       0.8       0.8
2         greatmusic          0.8          0.8       2.4       0.8
3         rockballad          0.7          0.7       0.7       0.7
4         amonamarth          0.3          0.3       0.3       0.3

In [101]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5290 entries, 0 to 5289
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   hashtag      5290 non-null   object 
 1   vader_score  3871 non-null   float64
 2   afinn_score  4532 non-null   float64
 3   ol_score     4831 non-null   float64
 4   ss_score     4471 non-null   float64
dtypes: float64(4), object(1)
memory usage: 206.8+ KB


In [102]:
## supprimer les lignes dans lesquelles il y'a au moins un NaN
df1 = df1.dropna(axis = 0, how ='any') 
df1.info()

<class 'pandas.core.frame.DataFrame'>
Index: 3868 entries, 0 to 5287
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   hashtag      3868 non-null   object 
 1   vader_score  3868 non-null   float64
 2   afinn_score  3868 non-null   float64
 3   ol_score     3868 non-null   float64
 4   ss_score     3868 non-null   float64
dtypes: float64(4), object(1)
memory usage: 181.3+ KB


In [103]:
# df1.sort_values(by='ol_score', ascending=False) # option 1
df1.sort_values(by=['ol_score', 'vader_score', 'afinn_score', 'ss_score'], ascending=False) ## option 2

hashtag  vader_score  afinn_score  ol_score  ss_score
4889          thriller          0.7          0.9       4.8  0.800000
1287     harmonicalove          0.6          0.8       4.7  0.671429
4174              well          0.6          0.7       3.9  0.650000
4458       richardmarx          0.6          0.6       3.6  0.600000
4924          richmond          0.6          0.6       3.6  0.600000
...                ...          ...          ...       ...       ...
4792          fearfits          0.1          0.1       0.1  0.100000
4841  pancreaticcancer          0.1          0.1       0.1  0.100000
4961        godhatesus          0.1          0.1       0.1  0.100000
5038     invisiblehate          0.1          0.1       0.1  0.100000
5268     blazesomehate          0.1          0.1       0.1  0.100000

[3868 rows x 5 columns]

In [104]:
#Rename column ol_score 
#df1 = df1.rename(columns = {'ol_score' : 'sentiment_score'}) option 1
df1.head()

hashtag  vader_score  afinn_score  ol_score  ss_score
0          relaxtime          0.8          0.8       2.4       0.8
1  melovechilicheese          0.8          0.8       0.8       0.8
2         greatmusic          0.8          0.8       2.4       0.8
3         rockballad          0.7          0.7       0.7       0.7
4         amonamarth          0.3          0.3       0.3       0.3

In [107]:
# count = df1.groupby(['sentiment_score']).count()  # option1
count = df1.groupby(['vader_score', 'afinn_score', 'ol_score', 'ss_score']).count() # option 2

count

hashtag
vader_score afinn_score ol_score ss_score         
0.1         0.1000      0.1      0.10           42
                        0.2      0.10           51
                        0.3      0.10           46
            0.2000      0.3      0.15            1
                        0.6      0.15            1
...                                            ...
0.9         0.9000      1.8      0.90           31
                        2.7      0.90            9
1.0         0.8875      1.0      1.00            1
            1.0000      2.0      1.00            1
                        3.0      1.00            2

[113 rows x 1 columns]

In [108]:
FILE_PATH = DATA_DIR / 'user_track_hashtag_timestamp.csv'
df2 = pd.read_csv(FILE_PATH)
df2.shape

(17560113, 4)

In [109]:
df2.head()

user_id                          track_id     hashtag   
0    81496937  cd52b3e5b51da29e5893dba82a418a4b  nowplaying  \
1    81496937  cd52b3e5b51da29e5893dba82a418a4b        goth   
2    81496937  cd52b3e5b51da29e5893dba82a418a4b   deathrock   
3    81496937  cd52b3e5b51da29e5893dba82a418a4b    postpunk   
4  2205686924  da3110a77b724072b08f231c9d6f7534  NowPlaying   

            created_at  
0  2014-01-01 05:54:21  
1  2014-01-01 05:54:21  
2  2014-01-01 05:54:21  
3  2014-01-01 05:54:21  
4  2014-01-01 05:54:22

In [110]:
df2.isna().sum()

user_id       0
track_id      0
hashtag       1
created_at    0
dtype: int64

In [111]:
df2 = df2.dropna(subset=['hashtag'])
df2.isna().sum()

user_id       0
track_id      0
hashtag       0
created_at    0
dtype: int64

In [113]:
# Get the count of the track_id
counts = df2['track_id'].value_counts()

# Drop track_id count < 50
df2 = df2[~df2['track_id'].isin(counts[counts < 50].index)]
df2.info()

<class 'pandas.core.frame.DataFrame'>
Index: 14966614 entries, 0 to 17560112
Data columns (total 4 columns):
 #   Column      Dtype 
---  ------      ----- 
 0   user_id     int64 
 1   track_id    object
 2   hashtag     object
 3   created_at  object
dtypes: int64(1), object(3)
memory usage: 570.9+ MB


In [17]:
df2.user_id.value_counts()

15518784      760846
2293477190    199314
318753478     185412
624797870     141574
1262924646    115376
               ...  
144367888          1
64057544           1
18417673           1
1567289976         1
1600836144         1
Name: user_id, Length: 126887, dtype: int64

In [18]:
df2.track_id.value_counts()

e24deb4963769d25e2c03ad9bf58a028    82551
3d02f9fcad37e6bb227682761039498c    43456
d6b1124fdd64c4b1afa59f967397111d    32735
5758909ef03fc3a2efaa57408ad43f22    32231
acca7456f99b4699871d3fe249e5fe11    32066
                                    ...  
d4ac12c1c07ddf3eb5aeb0ef3fdbf4de       50
a8d66b88b37b575b47e1f8f3bec68cee       50
f3bcbfbc16f3b5aac574216a88434c86       50
a7d341a3d7271f6b42b0c8cbdb909742       50
65c0ae1de781ebf202ba72a3a23b69fb       50
Name: track_id, Length: 49601, dtype: int64

In [20]:
#Merge CSV files into a single file based on hashtag
df_sentiment = pd.merge(df1, df2, on="hashtag", how='inner')
df_sentiment.head()

hashtag  sentiment_score    user_id                          track_id  \
0   relaxtime              2.4  104415394  9175ac1532ee7dbe97602866efabac58   
1   relaxtime              2.4  240771401  c2da30eb3450e8a3e5bfa16e8fa527da   
2   relaxtime              2.4  637256774  dcbb5aff8f96a79be9f59bc0e7b5c38d   
3  greatmusic              2.4  252330820  8f2ac86abb8bd48273c8fc95b632e347   
4  greatmusic              2.4   29235188  45bbd6d7cd65dc77596af8c5c0b89a70   

            created_at  
0  2014-05-20 07:24:40  
1  2014-10-12 22:46:57  
2  2014-11-07 13:01:08  
3  2014-02-13 16:18:51  
4  2014-05-15 20:38:46

In [21]:
df_sentiment.isna().sum()

hashtag            0
sentiment_score    0
user_id            0
track_id           0
created_at         0
dtype: int64

In [23]:
df_sentiment.shape

(5126717, 5)

In [24]:
df_sentiment['hashtag'].value_counts().head(10)

nowplaying         4870436
kiss92               56437
postpunk             25299
punk                 24730
deathrock            24699
urbantraxxradio      23867
tophits              22836
craveradio            9735
rock                  9483
stonerrock            4297
Name: hashtag, dtype: int64

#### Now that the two CSV files are joined (inner join), the new dataframe df_sentiment is reduced to 5,126,717 rows (from 17,560,114).

In [27]:
FILE_PATH = DATA_DIR / 'context_content_features.csv'
df3 = pd.read_csv(FILE_PATH, usecols=range(0,22))
df3.shape

(11614671, 22)

In [28]:
# Get the count of the track_id
counts = df3['track_id'].value_counts()

# Select the items where the track_id count is less than 50 and remove them
df3 = df3[~df3['track_id'].isin(counts[counts < 50].index)]

df3.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9143294 entries, 1 to 11614670
Data columns (total 22 columns):
 #   Column            Dtype  
---  ------            -----  
 0   coordinates       object 
 1   instrumentalness  float64
 2   liveness          float64
 3   speechiness       float64
 4   danceability      float64
 5   valence           float64
 6   loudness          float64
 7   tempo             float64
 8   acousticness      float64
 9   energy            float64
 10  mode              float64
 11  key               float64
 12  artist_id         object 
 13  place             object 
 14  geo               object 
 15  tweet_lang        object 
 16  track_id          object 
 17  created_at        object 
 18  lang              object 
 19  time_zone         object 
 20  user_id           float64
 21  id                int64  
dtypes: float64(12), int64(1), object(9)
memory usage: 1.6+ GB


In [29]:
#Drop unnecessary columns before merging with df_sentiment dataframe
df3 = df3.drop(['coordinates','id','place','geo'], axis=1)
df3.head()

instrumentalness  liveness  speechiness  danceability  valence  loudness  \
1          0.017700    0.0638       0.0624         0.769    0.752    -8.252   
2          0.000000    0.0860       0.0436         0.675    0.775    -4.432   
3          0.000000    0.1430       0.0292         0.324    0.333    -5.647   
5          0.000000    0.1100       0.0375         0.641    0.912    -4.271   
6          0.000006    0.3620       0.1340         0.554    0.677    -4.271   

     tempo  acousticness  energy  mode  key                         artist_id  \
1   95.862        0.2670   0.826   1.0  7.0  5cddcd0e314e2f2223ab21937d2c8778   
2   97.030        0.2170   0.885   0.0  1.0  e41273f43af504714d85465294f1f369   
3   74.101        0.2390   0.574   1.0  7.0  557ce373bd29743eb00a3723ab19ebe8   
5   93.010        0.0268   0.787   1.0  0.0  f965ec352eb8c0efc0af46244754942f   
6  126.045        0.0216   0.878   1.0  3.0  a3abd1e016fdba82a91379b7e2b9ab69   

  tweet_lang                          track_id           created_at lang  \
1         en  da3110a77b724072b08f231c9d6f7534  2014-01-01 05:54:22   en   
2         en  ba84d88c10fb0e42d4754a27ead10546  2014-01-01 05:54:22   es   
3         en  33f95122281f76e7134f9cbea3be980f  2014-01-01 05:54:24   en   
5         en  8bd5206b84c968eda0af8bc86d6ab1d1  2014-01-01 05:54:25   en   
6         en  23ced06ca57d37fa749b1595bc7ed1a4  2014-01-01 05:54:28   en   

                     time_zone       user_id  
1                          NaN  2.205687e+09  
2  Mountain Time (US & Canada)  1.325884e+08  
3   Eastern Time (US & Canada)  9.767522e+07  
5   Central Time (US & Canada)  4.522857e+08  
6                        Quito  6.508628e+07

In [32]:
df3 = df3.dropna()
df3['mode'] = df3['mode'].astype('Int64')
df3.shape
df3.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 6413576 entries, 2 to 11614670
Data columns (total 18 columns):
 #   Column            Dtype  
---  ------            -----  
 0   instrumentalness  float64
 1   liveness          float64
 2   speechiness       float64
 3   danceability      float64
 4   valence           float64
 5   loudness          float64
 6   tempo             float64
 7   acousticness      float64
 8   energy            float64
 9   mode              Int64  
 10  key               float64
 11  artist_id         object 
 12  tweet_lang        object 
 13  track_id          object 
 14  created_at        object 
 15  lang              object 
 16  time_zone         object 
 17  user_id           float64
dtypes: Int64(1), float64(11), object(6)
memory usage: 935.8+ MB


In [33]:
df3['lang'].unique()

array(['es', 'en', 'it', 'ja', 'pt', 'nl', 'id', 'en-gb', 'fr', 'ru',
       'de', 'ar', 'ko', 'pl', 'fi', 'ca', 'sv', 'tr', 'gl', 'fa', 'msa',
       'xx-lc', 'cs', 'da', 'zh-tw', 'eu', 'no', 'th', 'el', 'hu',
       'zh-cn', 'uk', 'ro', 'he', 'fil'], dtype=object)

In [49]:
display(df3['lang'].value_counts(normalize=True)*100)
display(df3['tweet_lang'].value_counts(normalize=True)*100)

en       86.235285
ja        4.381943
nl        3.436853
es        1.689543
de        1.351801
fr        0.993773
en-gb     0.611334
ru        0.324003
pt        0.269887
it        0.236252
id        0.209444
tr        0.113834
ko        0.077190
th        0.017026
pl        0.009029
uk        0.008053
fi        0.006967
sv        0.005824
hu        0.003623
ca        0.003539
xx-lc     0.003511
zh-tw     0.002954
cs        0.001979
no        0.001421
fil       0.001087
eu        0.001003
fa        0.000808
zh-cn     0.000585
msa       0.000502
el        0.000390
ar        0.000279
da        0.000167
gl        0.000056
he        0.000028
ro        0.000028
Name: lang, dtype: float64

en     92.163342
es      1.143862
ja      0.831674
in      0.825209
et      0.627302
ht      0.470302
fr      0.461580
de      0.375612
it      0.349111
sk      0.318820
pt      0.312355
pl      0.243636
tl      0.214432
no      0.205292
cy      0.198326
nl      0.184364
ro      0.179237
id      0.167617
fi      0.108484
da      0.105307
zh      0.098898
sv      0.092600
bs      0.053727
lt      0.049965
tr      0.041744
hu      0.038233
sl      0.025999
ko      0.025358
hr      0.023882
is      0.018420
ru      0.017277
vi      0.009781
lv      0.008109
bg      0.003483
und     0.001923
fa      0.001811
th      0.001616
uk      0.000947
ar      0.000334
el      0.000028
Name: tweet_lang, dtype: float64

In [53]:
df3['time_zone'].nunique()
df3['time_zone'].unique()
df3['time_zone'].value_counts(normalize=True)*100

Eastern Time (US & Canada)    29.203842
Central Time (US & Canada)    16.007073
Pacific Time (US & Canada)    15.070398
London                         4.402146
Beijing                        4.363523
                                ...    
PST                            0.000028
Samoa                          0.000028
America/El_Salvador            0.000028
Vladivostok                    0.000028
Europe/Belgrade                0.000028
Name: time_zone, Length: 164, dtype: float64

In [54]:
##Limit dataset to only en (English) language
df3 = df3.loc[~((df3['lang'] != 'en')),:]
df3.info()




<class 'pandas.core.frame.DataFrame'>
Int64Index: 4916702 entries, 3 to 11614670
Data columns (total 18 columns):
 #   Column            Dtype  
---  ------            -----  
 0   instrumentalness  float64
 1   liveness          float64
 2   speechiness       float64
 3   danceability      float64
 4   valence           float64
 5   loudness          float64
 6   tempo             float64
 7   acousticness      float64
 8   energy            float64
 9   mode              Int64  
 10  key               float64
 11  artist_id         object 
 12  tweet_lang        object 
 13  track_id          object 
 14  created_at        object 
 15  lang              object 
 16  time_zone         object 
 17  user_id           float64
dtypes: Int64(1), float64(11), object(6)
memory usage: 717.4+ MB


In [55]:
df3['lang'].unique()

array(['en'], dtype=object)

In [56]:
df4 = df_sentiment.merge(df3, on=['track_id','created_at','user_id'], how='inner')
df4.head()

hashtag  sentiment_score    user_id                          track_id  \
0  greatmusic              2.4  252330820  8f2ac86abb8bd48273c8fc95b632e347   
1  greatmusic              2.4   29235188  45bbd6d7cd65dc77596af8c5c0b89a70   
2  nowplaying              0.6   29235188  45bbd6d7cd65dc77596af8c5c0b89a70   
3     classic              1.0   29235188  45bbd6d7cd65dc77596af8c5c0b89a70   
4  greatmusic              2.4  195485165  6d45097acaaf6ed5ee55041f53249fa0   

            created_at  instrumentalness  liveness  speechiness  danceability  \
0  2014-02-13 16:18:51           0.00000    0.1980       0.1690         0.461   
1  2014-05-15 20:38:46           0.00001    0.0454       0.0451         0.677   
2  2014-05-15 20:38:46           0.00001    0.0454       0.0451         0.677   
3  2014-05-15 20:38:46           0.00001    0.0454       0.0451         0.677   
4  2014-10-22 18:36:54           0.00000    0.0769       0.0427         0.591   

   valence  loudness    tempo  acousticness  energy  mode   key  \
0    0.502    -9.046  143.055        0.1000   0.628     0   4.0   
1    0.886   -13.237   77.837        0.0313   0.544     0  11.0   
2    0.886   -13.237   77.837        0.0313   0.544     0  11.0   
3    0.886   -13.237   77.837        0.0313   0.544     0  11.0   
4    0.422    -6.315  100.063        0.0171   0.746     1  11.0   

                          artist_id tweet_lang lang  \
0  1a17d5f8af99355edd8a92679a02cb0d         en   en   
1  4a4e2ab094a4521b06252e9fdaf1fd0a         en   en   
2  4a4e2ab094a4521b06252e9fdaf1fd0a         en   en   
3  4a4e2ab094a4521b06252e9fdaf1fd0a         en   en   
4  f78fb5a7ddce990521f685522f3f8fce         en   en   

                    time_zone  
0  Eastern Time (US & Canada)  
1  Eastern Time (US & Canada)  
2  Eastern Time (US & Canada)  
3  Eastern Time (US & Canada)  
4                      London

In [57]:
#Convert hashtag info string
df4['hashtag'] = df4['hashtag'].astype(str)

#Convert user_id info string
df4['user_id'] = df4['user_id'].astype(str)

#Show changes
df4.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3094592 entries, 0 to 3094591
Data columns (total 20 columns):
 #   Column            Dtype  
---  ------            -----  
 0   hashtag           object 
 1   sentiment_score   float64
 2   user_id           object 
 3   track_id          object 
 4   created_at        object 
 5   instrumentalness  float64
 6   liveness          float64
 7   speechiness       float64
 8   danceability      float64
 9   valence           float64
 10  loudness          float64
 11  tempo             float64
 12  acousticness      float64
 13  energy            float64
 14  mode              Int64  
 15  key               float64
 16  artist_id         object 
 17  tweet_lang        object 
 18  lang              object 
 19  time_zone         object 
dtypes: Int64(1), float64(11), object(8)
memory usage: 498.8+ MB


In [58]:
#Create new column sentiment that will be the predictor based on the sentiment_score values
df4['sentiment'] = np.where(df4['sentiment_score']>= 0.01, 1, 0)
df4.head()

hashtag  sentiment_score    user_id                          track_id  \
0  greatmusic              2.4  252330820  8f2ac86abb8bd48273c8fc95b632e347   
1  greatmusic              2.4   29235188  45bbd6d7cd65dc77596af8c5c0b89a70   
2  nowplaying              0.6   29235188  45bbd6d7cd65dc77596af8c5c0b89a70   
3     classic              1.0   29235188  45bbd6d7cd65dc77596af8c5c0b89a70   
4  greatmusic              2.4  195485165  6d45097acaaf6ed5ee55041f53249fa0   

            created_at  instrumentalness  liveness  speechiness  danceability  \
0  2014-02-13 16:18:51           0.00000    0.1980       0.1690         0.461   
1  2014-05-15 20:38:46           0.00001    0.0454       0.0451         0.677   
2  2014-05-15 20:38:46           0.00001    0.0454       0.0451         0.677   
3  2014-05-15 20:38:46           0.00001    0.0454       0.0451         0.677   
4  2014-10-22 18:36:54           0.00000    0.0769       0.0427         0.591   

   valence  ...    tempo  acousticness  energy  mode   key  \
0    0.502  ...  143.055        0.1000   0.628     0   4.0   
1    0.886  ...   77.837        0.0313   0.544     0  11.0   
2    0.886  ...   77.837        0.0313   0.544     0  11.0   
3    0.886  ...   77.837        0.0313   0.544     0  11.0   
4    0.422  ...  100.063        0.0171   0.746     1  11.0   

                          artist_id tweet_lang lang  \
0  1a17d5f8af99355edd8a92679a02cb0d         en   en   
1  4a4e2ab094a4521b06252e9fdaf1fd0a         en   en   
2  4a4e2ab094a4521b06252e9fdaf1fd0a         en   en   
3  4a4e2ab094a4521b06252e9fdaf1fd0a         en   en   
4  f78fb5a7ddce990521f685522f3f8fce         en   en   

                    time_zone sentiment  
0  Eastern Time (US & Canada)         1  
1  Eastern Time (US & Canada)         1  
2  Eastern Time (US & Canada)         1  
3  Eastern Time (US & Canada)         1  
4                      London         1  

[5 rows x 21 columns]

In [61]:
#Drop all null value rows
df4 = df4.dropna()

#Drop unnecessary columns lang and created_at
df4 = df4.drop(['created_at','tweet_lang','lang'], axis=1)

df4.head()

hashtag  sentiment_score    user_id                          track_id  \
0  greatmusic              2.4  252330820  8f2ac86abb8bd48273c8fc95b632e347   
1  greatmusic              2.4   29235188  45bbd6d7cd65dc77596af8c5c0b89a70   
2  nowplaying              0.6   29235188  45bbd6d7cd65dc77596af8c5c0b89a70   
3     classic              1.0   29235188  45bbd6d7cd65dc77596af8c5c0b89a70   
4  greatmusic              2.4  195485165  6d45097acaaf6ed5ee55041f53249fa0   

   instrumentalness  liveness  speechiness  danceability  valence  loudness  \
0           0.00000    0.1980       0.1690         0.461    0.502    -9.046   
1           0.00001    0.0454       0.0451         0.677    0.886   -13.237   
2           0.00001    0.0454       0.0451         0.677    0.886   -13.237   
3           0.00001    0.0454       0.0451         0.677    0.886   -13.237   
4           0.00000    0.0769       0.0427         0.591    0.422    -6.315   

     tempo  acousticness  energy  mode   key  \
0  143.055        0.1000   0.628     0   4.0   
1   77.837        0.0313   0.544     0  11.0   
2   77.837        0.0313   0.544     0  11.0   
3   77.837        0.0313   0.544     0  11.0   
4  100.063        0.0171   0.746     1  11.0   

                          artist_id                   time_zone  sentiment  
0  1a17d5f8af99355edd8a92679a02cb0d  Eastern Time (US & Canada)          1  
1  4a4e2ab094a4521b06252e9fdaf1fd0a  Eastern Time (US & Canada)          1  
2  4a4e2ab094a4521b06252e9fdaf1fd0a  Eastern Time (US & Canada)          1  
3  4a4e2ab094a4521b06252e9fdaf1fd0a  Eastern Time (US & Canada)          1  
4  f78fb5a7ddce990521f685522f3f8fce                      London          1

In [62]:
#Look at unique values for time_zone column
df4.time_zone.unique()

array(['Eastern Time (US & Canada)', 'London', 'Monterrey', 'Alaska',
       'Amsterdam', 'Dublin', 'Central Time (US & Canada)',
       'Pacific Time (US & Canada)', 'Quito', 'Central America',
       'Buenos Aires', 'Mountain Time (US & Canada)', 'Jakarta',
       'Casablanca', 'Mexico City', 'Bangkok', 'Atlantic Time (Canada)',
       'Hawaii', 'Kuala Lumpur', 'Greenland', 'Santiago', 'Madrid',
       'Stockholm', 'Arizona', 'Saskatchewan', 'Beijing', 'Warsaw',
       'Lima', 'Edinburgh', 'Perth', 'Istanbul', 'Lisbon', 'Abu Dhabi',
       'La Paz', 'West Central Africa', 'Athens', 'America/Chicago',
       'Baghdad', 'Singapore', 'Budapest', 'Paris', 'Bern', 'Pretoria',
       'Brisbane', 'Brasilia', 'Moscow', 'Berlin', 'Kyiv', 'Hong Kong',
       'Helsinki', 'New Delhi', 'Sydney', 'Melbourne', 'Harare',
       'Yerevan', 'New Caledonia', 'Adelaide', 'Wellington', 'Copenhagen',
       'Yakutsk', 'Canberra', 'Rome', 'Tokyo', 'Chennai', 'Kathmandu',
       'Seoul', 'Auckland', 'St. Pe

In [63]:
#Make all USA Timezone values consistent
df4['time_zone'].replace('Eastern Time (US & Canada)', 'Eastern Time',inplace=True)
df4['time_zone'].replace('Central Time (US & Canada)', 'Central Time',inplace=True)
df4['time_zone'].replace('Pacific Time (US & Canada)', 'Pacific Time',inplace=True)
df4['time_zone'].replace('Mountain Time (US & Canada)', 'Mountain Time',inplace=True)
df4['time_zone'].replace('Alaska', 'Alaska Time',inplace=True)
df4['time_zone'].replace('Hawaii', 'Hawaii Time',inplace=True)
df4['time_zone'].replace('Arizona', 'Mountain Time',inplace=True)
df4['time_zone'].replace('America/Chicago', 'Central Time',inplace=True)
df4['time_zone'].replace('America/New_York', 'Eastern Time',inplace=True)
df4['time_zone'].replace('America/Los_Angeles', 'Pacific Time',inplace=True)
df4['time_zone'].replace('America/Denver', 'Mountain Time',inplace=True)
df4['time_zone'].replace('America/Detroit', 'Eastern Time',inplace=True)

In [64]:
#Limit dataset to only USA time zones
df4 = df4.loc[~((df4['time_zone'] != 'Eastern Time') & (df4['time_zone'] != 'Central Time') & (df4['time_zone'] != 'Pacific Time') & 
               (df4['time_zone'] != 'Mountain Time') & (df4['time_zone'] != 'Alaska Time') & (df4['time_zone'] != 'Hawaii Time')),:]
df4.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2267492 entries, 0 to 3094591
Data columns (total 18 columns):
 #   Column            Dtype  
---  ------            -----  
 0   hashtag           object 
 1   sentiment_score   float64
 2   user_id           object 
 3   track_id          object 
 4   instrumentalness  float64
 5   liveness          float64
 6   speechiness       float64
 7   danceability      float64
 8   valence           float64
 9   loudness          float64
 10  tempo             float64
 11  acousticness      float64
 12  energy            float64
 13  mode              Int64  
 14  key               float64
 15  artist_id         object 
 16  time_zone         object 
 17  sentiment         int32  
dtypes: Int64(1), float64(11), int32(1), object(5)
memory usage: 322.2+ MB


In [65]:
df4['sentiment'] = df4['sentiment'].astype('Int64')
df4.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2267492 entries, 0 to 3094591
Data columns (total 18 columns):
 #   Column            Dtype  
---  ------            -----  
 0   hashtag           object 
 1   sentiment_score   float64
 2   user_id           object 
 3   track_id          object 
 4   instrumentalness  float64
 5   liveness          float64
 6   speechiness       float64
 7   danceability      float64
 8   valence           float64
 9   loudness          float64
 10  tempo             float64
 11  acousticness      float64
 12  energy            float64
 13  mode              Int64  
 14  key               float64
 15  artist_id         object 
 16  time_zone         object 
 17  sentiment         Int64  
dtypes: Int64(2), float64(11), object(5)
memory usage: 333.0+ MB


In [66]:
## DESTINATION_PATH = DATA_DIR.parent / 'cleaned_dataset.csv'
## df4.to_csv(DESTINATION_PATH)

## artist_id and "hashtag were not dropped" so 5 dtypes object instead of 3
#Reduced the dataset from 4,916,702 to 2,267,492 by limiting it to USA Timezone data only.
## Create MVP Dataset minimum viable product


In [67]:
# Reorder columns
df_mvp = df4[['sentiment','sentiment_score', 'hashtag', 'user_id','track_id','artist_id','time_zone','instrumentalness',
              'liveness','speechiness','danceability','valence','loudness','tempo','acousticness','energy','mode','key']]
df_mvp.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2267492 entries, 0 to 3094591
Data columns (total 18 columns):
 #   Column            Dtype  
---  ------            -----  
 0   sentiment         Int64  
 1   sentiment_score   float64
 2   hashtag           object 
 3   user_id           object 
 4   track_id          object 
 5   artist_id         object 
 6   time_zone         object 
 7   instrumentalness  float64
 8   liveness          float64
 9   speechiness       float64
 10  danceability      float64
 11  valence           float64
 12  loudness          float64
 13  tempo             float64
 14  acousticness      float64
 15  energy            float64
 16  mode              Int64  
 17  key               float64
dtypes: Int64(2), float64(11), object(5)
memory usage: 333.0+ MB


In [68]:
# Create new dataset with only user_id, track_id and time_zone and the category variables
df_timezone = df_mvp.drop(['user_id','track_id', 'hashtag', 'artist_id', 'user_id','sentiment','sentiment_score','instrumentalness','liveness','speechiness',
                             'danceability','valence','loudness','tempo','acousticness','energy','mode','key'], axis=1)
df_timezone.head()

time_zone
0  Eastern Time
1  Eastern Time
2  Eastern Time
3  Eastern Time
7   Alaska Time

In [71]:
from sklearn.preprocessing import LabelEncoder
from sklearn.preprocessing import OneHotEncoder

# One Hot Encode the category data set
one_hot_df = pd.get_dummies(df_timezone)
one_hot_df.head()

time_zone_Alaska Time  time_zone_Central Time  time_zone_Eastern Time  \
0                      0                       0                       1   
1                      0                       0                       1   
2                      0                       0                       1   
3                      0                       0                       1   
7                      1                       0                       0   

   time_zone_Hawaii Time  time_zone_Mountain Time  time_zone_Pacific Time  
0                      0                        0                       0  
1                      0                        0                       0  
2                      0                        0                       0  
3                      0                        0                       0  
7                      0                        0                       0

In [72]:
# Rename the columns
one_hot_df.columns = ['tz_Alaska_Time','tz_Central_Time','tz_Eastern_Time','tz_Hawaii_Time','tz_Mountain_Time','tz_Pacific_Time']
one_hot_df.head()

tz_Alaska_Time  tz_Central_Time  tz_Eastern_Time  tz_Hawaii_Time  \
0               0                0                1               0   
1               0                0                1               0   
2               0                0                1               0   
3               0                0                1               0   
7               1                0                0               0   

   tz_Mountain_Time  tz_Pacific_Time  
0                 0                0  
1                 0                0  
2                 0                0  
3                 0                0  
7                 0                0

In [73]:
#Drop time_zone from MVP dataset
df_mvp = df_mvp.drop(['time_zone'], axis=1)
df_mvp.columns

Index(['sentiment', 'sentiment_score', 'hashtag', 'user_id', 'track_id',
       'artist_id', 'instrumentalness', 'liveness', 'speechiness',
       'danceability', 'valence', 'loudness', 'tempo', 'acousticness',
       'energy', 'mode', 'key'],
      dtype='object')

In [74]:
#Concatenate one hot encoded dataframe
df_mvp = pd.concat([df_mvp,one_hot_df], axis=1)
df_mvp.head()

sentiment  sentiment_score     hashtag    user_id  \
0          1              2.4  greatmusic  252330820   
1          1              2.4  greatmusic   29235188   
2          1              0.6  nowplaying   29235188   
3          1              1.0     classic   29235188   
7          1              0.3  amonamarth  124052165   

                           track_id                         artist_id  \
0  8f2ac86abb8bd48273c8fc95b632e347  1a17d5f8af99355edd8a92679a02cb0d   
1  45bbd6d7cd65dc77596af8c5c0b89a70  4a4e2ab094a4521b06252e9fdaf1fd0a   
2  45bbd6d7cd65dc77596af8c5c0b89a70  4a4e2ab094a4521b06252e9fdaf1fd0a   
3  45bbd6d7cd65dc77596af8c5c0b89a70  4a4e2ab094a4521b06252e9fdaf1fd0a   
7  d53b2de022666067050dae8f6645edc2  942c9f2520684c22eb6216a92b711f9e   

   instrumentalness  liveness  speechiness  danceability  ...  acousticness  \
0          0.000000    0.1980       0.1690         0.461  ...      0.100000   
1          0.000010    0.0454       0.0451         0.677  ...      0.031300   
2          0.000010    0.0454       0.0451         0.677  ...      0.031300   
3          0.000010    0.0454       0.0451         0.677  ...      0.031300   
7          0.000142    0.1130       0.0709         0.355  ...      0.000004   

   energy  mode   key  tz_Alaska_Time  tz_Central_Time  tz_Eastern_Time  \
0   0.628     0   4.0               0                0                1   
1   0.544     0  11.0               0                0                1   
2   0.544     0  11.0               0                0                1   
3   0.544     0  11.0               0                0                1   
7   0.980     0   7.0               1                0                0   

   tz_Hawaii_Time  tz_Mountain_Time  tz_Pacific_Time  
0               0                 0                0  
1               0                 0                0  
2               0                 0                0  
3               0                 0                0  
7               0                 0                0  

[5 rows x 23 columns]

In [75]:
DESTINATION_PATH = DATA_DIR / 'cleaned_dataset.csv'
df_mvp.to_csv(DESTINATION_PATH)

## La suite n'est pas à utiliser pour l'instant

In [1]:
import os
from pathlib import Path
import pandas as pd

CURRENT_DIR = os.path.abspath('')
DATA_DIR = Path(CURRENT_DIR).parent / "data"


In [ ]:
## tentative de merge dataframe d'origine nettoyé avec dataframe MuSe
FILE_PATH = DATA_DIR/'muse_v3.csv'

df6 = pd.read_csv(FILE_PATH)
df6 = df6.dropna(axis=0, subset='mbid')
df6 = df6.drop(['lastfm_url', 'track', 'artist', 'spotify_id', 'genre'], axis=1)
df6['track_id'] = df6['mbid'].apply(lambda x: x[:8] + x[9:13] + x[14:18] + x[19:23] +x[24:])

In [ ]:
test_list = []
for i in df6['track_id']:
    test_list.append(i)


In [ ]:
df_mvp['track_id'].isin(test_list).sum()

In [2]:
## tentative de merge de MuSe avec https://www.kaggle.com/datasets/maharshipandya/-spotify-tracks-dataset 
FILE_PATH = DATA_DIR/'spotify_dataset.csv'

In [3]:
df_spotify = pd.read_csv(FILE_PATH)
display(df_spotify.head())
df_spotify.info()

Unnamed: 0                track_id                 artists   
0           0  5SuOikwiRyPMVoIQDJUgSV             Gen Hoshino  \
1           1  4qPNDBW1i3p13qLCt0Ki3A            Ben Woodward   
2           2  1iJBSr7s7jYXzM8EGcbK5b  Ingrid Michaelson;ZAYN   
3           3  6lfxq3CG4xtTiEg7opyCyx            Kina Grannis   
4           4  5vjLSffimiIP26QG5WcN2K        Chord Overstreet   

                                          album_name   
0                                             Comedy  \
1                                   Ghost (Acoustic)   
2                                     To Begin Again   
3  Crazy Rich Asians (Original Motion Picture Sou...   
4                                            Hold On   

                   track_name  popularity  duration_ms  explicit   
0                      Comedy          73       230666     False  \
1            Ghost - Acoustic          55       149610     False   
2              To Begin Again          57       210826     False   
3  Can't Help Falling In Love          71       201933     False   
4                     Hold On          82       198853     False   

   danceability  energy  ...  loudness  mode  speechiness  acousticness   
0         0.676  0.4610  ...    -6.746     0       0.1430        0.0322  \
1         0.420  0.1660  ...   -17.235     1       0.0763        0.9240   
2         0.438  0.3590  ...    -9.734     1       0.0557        0.2100   
3         0.266  0.0596  ...   -18.515     1       0.0363        0.9050   
4         0.618  0.4430  ...    -9.681     1       0.0526        0.4690   

   instrumentalness  liveness  valence    tempo  time_signature  track_genre  
0          0.000001    0.3580    0.715   87.917               4     acoustic  
1          0.000006    0.1010    0.267   77.489               4     acoustic  
2          0.000000    0.1170    0.120   76.332               4     acoustic  
3          0.000071    0.1320    0.143  181.740               3     acoustic  
4          0.000000    0.0829    0.167  119.949               4     acoustic  

[5 rows x 21 columns]

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 114000 entries, 0 to 113999
Data columns (total 21 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   Unnamed: 0        114000 non-null  int64  
 1   track_id          114000 non-null  object 
 2   artists           113999 non-null  object 
 3   album_name        113999 non-null  object 
 4   track_name        113999 non-null  object 
 5   popularity        114000 non-null  int64  
 6   duration_ms       114000 non-null  int64  
 7   explicit          114000 non-null  bool   
 8   danceability      114000 non-null  float64
 9   energy            114000 non-null  float64
 10  key               114000 non-null  int64  
 11  loudness          114000 non-null  float64
 12  mode              114000 non-null  int64  
 13  speechiness       114000 non-null  float64
 14  acousticness      114000 non-null  float64
 15  instrumentalness  114000 non-null  float64
 16  liveness          11

In [4]:
df_spotify = df_spotify.drop('Unnamed: 0', axis=1)
display(df_spotify.head())
df_spotify.info()

track_id                 artists   
0  5SuOikwiRyPMVoIQDJUgSV             Gen Hoshino  \
1  4qPNDBW1i3p13qLCt0Ki3A            Ben Woodward   
2  1iJBSr7s7jYXzM8EGcbK5b  Ingrid Michaelson;ZAYN   
3  6lfxq3CG4xtTiEg7opyCyx            Kina Grannis   
4  5vjLSffimiIP26QG5WcN2K        Chord Overstreet   

                                          album_name   
0                                             Comedy  \
1                                   Ghost (Acoustic)   
2                                     To Begin Again   
3  Crazy Rich Asians (Original Motion Picture Sou...   
4                                            Hold On   

                   track_name  popularity  duration_ms  explicit   
0                      Comedy          73       230666     False  \
1            Ghost - Acoustic          55       149610     False   
2              To Begin Again          57       210826     False   
3  Can't Help Falling In Love          71       201933     False   
4                     Hold On          82       198853     False   

   danceability  energy  key  loudness  mode  speechiness  acousticness   
0         0.676  0.4610    1    -6.746     0       0.1430        0.0322  \
1         0.420  0.1660    1   -17.235     1       0.0763        0.9240   
2         0.438  0.3590    0    -9.734     1       0.0557        0.2100   
3         0.266  0.0596    0   -18.515     1       0.0363        0.9050   
4         0.618  0.4430    2    -9.681     1       0.0526        0.4690   

   instrumentalness  liveness  valence    tempo  time_signature track_genre  
0          0.000001    0.3580    0.715   87.917               4    acoustic  
1          0.000006    0.1010    0.267   77.489               4    acoustic  
2          0.000000    0.1170    0.120   76.332               4    acoustic  
3          0.000071    0.1320    0.143  181.740               3    acoustic  
4          0.000000    0.0829    0.167  119.949               4    acoustic

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 114000 entries, 0 to 113999
Data columns (total 20 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   track_id          114000 non-null  object 
 1   artists           113999 non-null  object 
 2   album_name        113999 non-null  object 
 3   track_name        113999 non-null  object 
 4   popularity        114000 non-null  int64  
 5   duration_ms       114000 non-null  int64  
 6   explicit          114000 non-null  bool   
 7   danceability      114000 non-null  float64
 8   energy            114000 non-null  float64
 9   key               114000 non-null  int64  
 10  loudness          114000 non-null  float64
 11  mode              114000 non-null  int64  
 12  speechiness       114000 non-null  float64
 13  acousticness      114000 non-null  float64
 14  instrumentalness  114000 non-null  float64
 15  liveness          114000 non-null  float64
 16  valence           11

In [5]:
df_spotify.duplicated().sum()

450

In [6]:
df_spotify = df_spotify.drop_duplicates(keep='first')
print(df_spotify.duplicated().sum())
display(df_spotify.info())

0
<class 'pandas.core.frame.DataFrame'>
Index: 113550 entries, 0 to 113999
Data columns (total 20 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   track_id          113550 non-null  object 
 1   artists           113549 non-null  object 
 2   album_name        113549 non-null  object 
 3   track_name        113549 non-null  object 
 4   popularity        113550 non-null  int64  
 5   duration_ms       113550 non-null  int64  
 6   explicit          113550 non-null  bool   
 7   danceability      113550 non-null  float64
 8   energy            113550 non-null  float64
 9   key               113550 non-null  int64  
 10  loudness          113550 non-null  float64
 11  mode              113550 non-null  int64  
 12  speechiness       113550 non-null  float64
 13  acousticness      113550 non-null  float64
 14  instrumentalness  113550 non-null  float64
 15  liveness          113550 non-null  float64
 16  valence           11355

None

In [7]:
display(df_spotify.isna().sum())


track_id            0
artists             1
album_name          1
track_name          1
popularity          0
duration_ms         0
explicit            0
danceability        0
energy              0
key                 0
loudness            0
mode                0
speechiness         0
acousticness        0
instrumentalness    0
liveness            0
valence             0
tempo               0
time_signature      0
track_genre         0
dtype: int64

In [8]:
df_spotify = df_spotify.dropna()
df_spotify.info()

<class 'pandas.core.frame.DataFrame'>
Index: 113549 entries, 0 to 113999
Data columns (total 20 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   track_id          113549 non-null  object 
 1   artists           113549 non-null  object 
 2   album_name        113549 non-null  object 
 3   track_name        113549 non-null  object 
 4   popularity        113549 non-null  int64  
 5   duration_ms       113549 non-null  int64  
 6   explicit          113549 non-null  bool   
 7   danceability      113549 non-null  float64
 8   energy            113549 non-null  float64
 9   key               113549 non-null  int64  
 10  loudness          113549 non-null  float64
 11  mode              113549 non-null  int64  
 12  speechiness       113549 non-null  float64
 13  acousticness      113549 non-null  float64
 14  instrumentalness  113549 non-null  float64
 15  liveness          113549 non-null  float64
 16  valence           113549 

In [120]:
## Lost only 1 row: all NaNs were on the same observation

In [9]:
df_spotify['track_id'].nunique()

89740

In [10]:
## Tentative de merge avec le dataframe MuSe.csv 
## à ne pas utiliser pour l'instant


FILE_PATH = DATA_DIR/'muse_v3.csv'

df_muse = pd.read_csv(FILE_PATH)
df_muse

lastfm_url   
0      https://www.last.fm/music/eminem/_/%2527till%2...  \
1      https://www.last.fm/music/metallica/_/st.%2banger   
2      https://www.last.fm/music/rick%2bross/_/speedi...   
3      https://www.last.fm/music/m.i.a./_/bamboo%2bbanga   
4        https://www.last.fm/music/dope/_/die%2bmf%2bdie   
...                                                  ...   
89996  https://www.last.fm/music/gaby%2bhoffmann%2b%2...   
89997  https://www.last.fm/music/omar%2brodriguez-lop...   
89998  https://www.last.fm/music/ace%2bwhite/_/crookf...   
89999  https://www.last.fm/music/second%2bchyld/_/tra...   
90000  https://www.last.fm/music/message%2bto%2bbears...   

                                         track                       artist   
0                             'Till I Collapse                       Eminem  \
1                                    St. Anger                    Metallica   
2                                     Speedin'                    Rick Ross   
3                                 Bamboo Banga                       M.I.A.   
4                                   Die MF Die                         Dope   
...                                        ...                          ...   
89996                                   Battle  Gaby Hoffmann & Jay Duplass   
89997  Hands Tied to the Roots of a Hemorrhage         Omar Rodriguez-Lopez   
89998                           Crookfield Zoo                    Ace White   
89999               Transparent (Full Version)                 Second Chyld   
90000                                   Unfold             Message To Bears   

                                            seeds  number_of_emotion_tags   
0                                  ['aggressive']                       6  \
1                                  ['aggressive']                       8   
2                                  ['aggressive']                       1   
3      ['aggressive', 'fun', 'sexy', 'energetic']                      13   
4                                  ['aggressive']                       7   
...                                           ...                     ...   
89996                             ['transparent']                       1   
89997                             ['transparent']                       4   
89998                             ['transparent']                       1   
89999                             ['transparent']                       1   
90000                             ['translucent']                       2   

       valence_tags  arousal_tags  dominance_tags   
0          4.550000      5.273125        5.690625  \
1          3.710000      5.833000        5.427250   
2          3.080000      5.870000        5.490000   
3          6.555071      5.537214        5.691357   
4          3.771176      5.348235        5.441765   
...             ...           ...             ...   
89996      5.370000      3.450000        5.330000   
89997      5.797887      4.132254        5.570563   
89998      5.370000      3.450000        5.330000   
89999      5.370000      3.450000        5.330000   
90000      3.340000      1.405000        3.500000   

                                       mbid              spotify_id   
0      cab93def-26c5-4fb0-bedd-26ec4c1619e1  4xkOaSrkexMciUUogZKVTS  \
1      727a2529-7ee8-4860-aef6-7959884895cb  3fOc9x06lKJBhz435mInlH   
2                                       NaN  3Y96xd4Ce0J47dcalLrEC8   
3      99dd2c8c-e7c1-413e-8ea4-4497a00ffa18  6tqFC1DIOphJkCwrjVzPmg   
4      b9eb3484-5e0e-4690-ab5a-ca91937032a5  5bU4KX47KqtDKKaLM4QCzh   
...                                     ...                     ...   
89996                                   NaN                     NaN   
89997                                   NaN                     NaN   
89998                                   NaN                     NaN   
89999                                   NaN                     NaN   
90000  247179b0-af1f-4c2b-b4fe-75

In [11]:
print(f"NaNs in {df_muse['mbid'].isna().sum()} rows of mbid track key")
print(f"NaNs in {df_muse['spotify_id'].isna().sum()} rows of spotify_id track key")
print(f"NaNs in {df_muse['track'].isna().sum()} rows of 'track' track key")

NaNs in 28784 rows of mbid track key
NaNs in 28371 rows of spotify_id track key
NaNs in 0 rows of 'track' track key


In [13]:
track_list_muse = df_muse['track'].unique().tolist()
print(len(track_list_muse))
df_spotify['track_name'].isin(track_list_muse).sum()

79392


17439

In [15]:
df_muse = df_muse.drop(['lastfm_url'], axis=1)
df_muse

track                       artist   
0                             'Till I Collapse                       Eminem  \
1                                    St. Anger                    Metallica   
2                                     Speedin'                    Rick Ross   
3                                 Bamboo Banga                       M.I.A.   
4                                   Die MF Die                         Dope   
...                                        ...                          ...   
89996                                   Battle  Gaby Hoffmann & Jay Duplass   
89997  Hands Tied to the Roots of a Hemorrhage         Omar Rodriguez-Lopez   
89998                           Crookfield Zoo                    Ace White   
89999               Transparent (Full Version)                 Second Chyld   
90000                                   Unfold             Message To Bears   

                                            seeds  number_of_emotion_tags   
0                                  ['aggressive']                       6  \
1                                  ['aggressive']                       8   
2                                  ['aggressive']                       1   
3      ['aggressive', 'fun', 'sexy', 'energetic']                      13   
4                                  ['aggressive']                       7   
...                                           ...                     ...   
89996                             ['transparent']                       1   
89997                             ['transparent']                       4   
89998                             ['transparent']                       1   
89999                             ['transparent']                       1   
90000                             ['translucent']                       2   

       valence_tags  arousal_tags  dominance_tags   
0          4.550000      5.273125        5.690625  \
1          3.710000      5.833000        5.427250   
2          3.080000      5.870000        5.490000   
3          6.555071      5.537214        5.691357   
4          3.771176      5.348235        5.441765   
...             ...           ...             ...   
89996      5.370000      3.450000        5.330000   
89997      5.797887      4.132254        5.570563   
89998      5.370000      3.450000        5.330000   
89999      5.370000      3.450000        5.330000   
90000      3.340000      1.405000        3.500000   

                                       mbid              spotify_id   
0      cab93def-26c5-4fb0-bedd-26ec4c1619e1  4xkOaSrkexMciUUogZKVTS  \
1      727a2529-7ee8-4860-aef6-7959884895cb  3fOc9x06lKJBhz435mInlH   
2                                       NaN  3Y96xd4Ce0J47dcalLrEC8   
3      99dd2c8c-e7c1-413e-8ea4-4497a00ffa18  6tqFC1DIOphJkCwrjVzPmg   
4      b9eb3484-5e0e-4690-ab5a-ca91937032a5  5bU4KX47KqtDKKaLM4QCzh   
...                                     ...                     ...   
89996                                   NaN                     NaN   
89997                                   NaN                     NaN   
89998                                   NaN                     NaN   
89999                                   NaN                     NaN   
90000  247179b0-af1f-4c2b-b4fe-751070d1c0cc  14BYEOeRb3jzglJmKX8eM2   

                  genre  
0                   rap  
1                 metal  
2                   rap  
3               hip-hop  
4                 metal  
...                 ...  
89996               NaN  
89997  progressive rock  
89998               NaN  
89999               NaN  
90000           ambient  

[90001 rows x 10 columns]

In [91]:
df_muse = df_muse.rename(columns={'track' : 'track_name', 'artist': 'artists'})

In [92]:
df_spotify_muse = pd.merge(df_spotify, df_muse, on=['track_name', 'artists'])

In [93]:
df_spotify_muse

track_id           artists   
0     1EzrEOXmMH3G43AXT1y7pA        Jason Mraz  \
1     3S0OXQeoh0w6AY8WQVckRW        Jason Mraz   
2     03yc0G2OoH1Eeyu7Piy8fK        Jason Mraz   
3     1EzrEOXmMH3G43AXT1y7pA        Jason Mraz   
4     5ivF4eQBqJiVL5IAE9jRyl        Jason Mraz   
...                      ...               ...   
2176  5WGpg5FAO6OOm396QS53Re      Phil Wickham   
2177  7j1e730A8K1my1mM4knYmr      Chris Tomlin   
2178  0BP0JHEeYVZZsyngW4Uc1x      Chris Tomlin   
2179  3CO19cTl5zzlWx9GWEpOhD  Michael W. Smith   
2180  3stuliHpnexsTj6uXFJ5iI      Audrey Assad   

                                          album_name              track_name   
0                We Sing. We Dance. We Steal Things.               I'm Yours  \
1                We Sing. We Dance. We Steal Things.               I'm Yours   
2                                          I'm Yours               I'm Yours   
3                We Sing. We Dance. We Steal Things.               I'm Yours   
4                         Love Is a Four Letter Word         I Won't Give Up   
...                                              ...                     ...   
2176  You Are I Am - 6 Essential Christian Pop Songs        You're Beautiful   
2177  How Great Is Our God: The Essential Collection             I Will Rise   
2178                                      Hello Love             I Will Rise   
2179                               Change Your World  I Will Be Here For You   
2180                       The House You're Building                Restless   

      popularity  duration_ms  explicit  danceability  energy  key  ...   
0             80       242946     False         0.703   0.444   11  ...  \
1             75       242946     False         0.703   0.444   11  ...   
2             60       243494     False         0.693   0.430   11  ...   
3             80       242946     False         0.703   0.444   11  ...   
4             69       240165     False         0.483   0.303    4  ...   
...          ...          ...       ...           ...     ...  ...  ...   
2176          26       293720     False         0.329   0.518    3  ...   
2177          47       300373     False         0.213   0.320   11  ...   
2178          46       301240     False         0.438   0.320   11  ...   
2179          53       273773     False         0.425   0.519    9  ...   
2180          47       295064     False         0.422   0.377   11  ...   

      time_signature  track_genre                seeds   
0                  4     acoustic  ['happy', 'mellow']  \
1                  4     acoustic  ['happy', 'mellow']   
2                  4     acoustic  ['happy', 'mellow']   
3                  4         rock  ['happy', 'mellow']   
4                  3     acoustic         ['romantic']   
...              ...          ...                  ...   
2176               4  world-music      ['sentimental']   
2177               4  world-music         ['soothing']   
2178               4  world-music         ['soothing']   
2179               4  world-music        ['spiritual']   
2180               4  world-music            ['light']   

      number_of_emotion_tags  valence_tags  arousal_tags  dominance_tags   
0                         14      7.379128      4.824633        6.430642  \
1                         14      7.379128      4.824633        6.430642   
2                         14      7.379128      4.824633        6.430642   
3                         14      7.379128      4.824633        6.430642   
4                         11      7.655366      5.079024        6.090569   
...                      ...           ...           ...             ...   
2176                       3      5.240432      3.199568        4.783741   
2177                       3      5.416471      3.337647        4.507059   
2178                       3      5.416471      3.337647        4.507059   
2179                       4      6.135000      4.251765        5.960294   
2180                    

In [ ]:
## seulement 2181 entrées communes

In [ ]:
# Tentative de merge de MuSe avec large spotify dataset 6.10E3 songs https://www.kaggle.com/general/153757

In [16]:
FILE_PATH = DATA_DIR / 'large_spotify_tracks.csv'
df_test = pd.read_csv(FILE_PATH)

In [19]:
df_test.head()

id                                 name  popularity   
0  35iwgR4jXetI318WEWsa1Q                                Carve           6  \
1  021ht4sdgPcrDgSk7JTbKY  Capítulo 2.16 - Banquero Anarquista           0   
2  07A5yehtSnoedViJAZkNnc   Vivo para Quererte - Remasterizado           0   
3  08FmqUhxtyLTn6pAh6bk45        El Prisionero - Remasterizado           0   
4  08y9GfoqCWfOGsKdwojr5e                  Lady of the Evening           0   

   duration_ms  explicit              artists                  id_artists   
0       126903         0              ['Uli']  ['45tIt06XoI0Iio4LBEVpls']  \
1        98200         0  ['Fernando Pessoa']  ['14jtPCOoNZwquk5wd9DxrY']   
2       181640         0  ['Ignacio Corsini']  ['5LiOoJbxVSAMkBS2fUm3X2']   
3       176907         0  ['Ignacio Corsini']  ['5LiOoJbxVSAMkBS2fUm3X2']   
4       163080         0      ['Dick Haymes']  ['3BiJGZsyX9sJchTqcSA7Su']   

  release_date  danceability  energy  key  loudness  mode  speechiness   
0   1922-02-22         0.645  0.4450    0   -13.338     1       0.4510  \
1   1922-06-01         0.695  0.2630    0   -22.136     1       0.9570   
2   1922-03-21         0.434  0.1770    1   -21.180     1       0.0512   
3   1922-03-21         0.321  0.0946    7   -27.961     1       0.0504   
4         1922         0.402  0.1580    3   -16.900     0       0.0390   

   acousticness  instrumentalness  liveness  valence    tempo  time_signature  
0         0.674            0.7440     0.151    0.127  104.851               3  
1         0.797            0.0000     0.148    0.655  102.009               1  
2         0.994            0.0218     0.212    0.457  130.418               5  
3         0.995            0.9180     0.104    0.397  169.980               3  
4         0.989            0.1300     0.311    0.196  103.220               4

In [25]:
df_test = df_test.sort_values(by='release_date', ascending=False)
date_list = df_test['release_date'].unique().tolist()

In [39]:
df_test2 = df_test[(df_test['release_date']) > '1950-01-01']

In [40]:
df_test2

id   
93919   6cjSDzbazK0wsWWbZMRxsl  \
188893  2romdyQNQM3AYoNWLc12wD   
188882  25GXWqZZO8A9Y6xcPOV1iv   
188881  24DMzkRASCMSJrT5swk27L   
94285   6tbdUtj5EDrJnIVfG2loQR   
...                        ...   
24297   0OcOeyGdreX6CJvP60pFHK   
546269  0NkXBqX1TQWgCtbBruTiLk   
24303   0Our1KG98d3Vm42nuEqQab   
24532   0hOGFf6P7lJvFe30yYuOe6   
119707  0j7JB3ToGKjABNEKzfxlit   

                                                     name  popularity   
93919                               Blow Your Mind (Mwah)           0  \
188893                                              Ahora           0   
188882                                              Negro           0   
188881                                              Verde           0   
94285                                             The Box           0   
...                                                   ...         ...   
24297                    Alexander the Great (1956) Suite           0   
546269  Groucho Marx, You Bet Your Life, Old Time Radi...           0   
24303   I Want to Follow the Swallow Back Home to Colo...           0   
24532                                Lane County Bachelor           0   
119707                                          Dolly Day           0   

        duration_ms  explicit                              artists   
93919        178583         0                         ['Dua Lipa']  \
188893       254907         0                         ['J Balvin']   
188882       182507         0                         ['J Balvin']   
188881       142973         0                  ['J Balvin', 'Sky']   
94285        196653         0                      ['Roddy Ricch']   
...             ...       ...                                  ...   
24297        390797         0                 ['Mario Nascimbene']   
546269      1716156         0  ['Groucho Marx', 'George Fenneman']   
24303        122664         0             ['Sons of the Pioneers']   
24532        116304         0             ['Sons of the Pioneers']   
119707       162432         0             ['Sons of the Pioneers']   

                                               id_artists release_date   
93919                          ['6M2wZ9GZgrQXHCFfjv46we']   2021-04-16  \
188893                         ['1vyhD5VmyZ7KMfW5gqLgo5']   2021-04-16   
188882                         ['1vyhD5VmyZ7KMfW5gqLgo5']   2021-04-16   
188881  ['1vyhD5VmyZ7KMfW5gqLgo5', '7hCaz22W6seMspHOPD...   2021-04-16   
94285                          ['757aE44tKEUQEqRuT6GnEB']   2021-04-16   
...                                                   ...          ...   
24297                          ['2sy2OHRJWSjjDUjQd9coa7']   1950-02-01   
546269  ['7hKik9RqoE5H7qTIesQMMf', '0zKqniuwfEYM1UvWN3...   1950-01-11   
24303                          ['1gfIkFZ4hIs2gETkRVaY68']   1950-01-10   
24532                          ['1gfIkFZ4hIs2gETkRVaY68']   1950-01-10   
119707                         ['1gfIkFZ4hIs2gETkRVaY68']   1950-01-10   

        danceability  energy  key  loudness  mode  speechiness  acousticness   
93919          0.677  0.7960    2    -4.761     0       0.1170        0.0188  \
188893         0.810  0.7840    3    -3.223     0       0.0622        0.0874   
188882         0.791  0.9070    0    -2.223     1       0.0884        0.0132   
188881         0.831  0.7410    8    -4.281     0       0.0508        0.0356   
94285          0.896  0.5860   10    -6.687     0       0.0559        0.1040   
...              ...     ...  ...       ...   ...          ...           ...   
24297          0.225  0.2440    7   -13.786     0       0.0318        0.8870   
546269         0.532  0.4810    0   -15.583     0       0.9560        0.8200   
24303          0.726  0.0931   10   -19.224     1       0.0575        0.9350   
24532          0.555  0.1440    2   -18.863     1       0.1270        0.9910   
119707         0.518  0.1400    3   -19.262     1       0.0478        0.9070   

        instrumentalness  liveness  valence    temp

In [44]:
track_list_muse = df_muse['track'].unique().tolist()
print(len(track_list_muse))
df_test2['name'].isin(track_list_muse).sum()

79392


52332

In [38]:
#df_muse[df_muse['track']=='Blow Your Mind (Mwah)']

track    artist   
3979  Blow Your Mind (Mwah)  Dua Lipa  \

                                               seeds  number_of_emotion_tags   
3979  ['harsh', 'dark', 'silly', 'sexy', 'negative']                      10  \

      valence_tags  arousal_tags  dominance_tags mbid              spotify_id   
3979      4.684474          5.07        4.979737  NaN  7kJlTKjNZVT26iwiDUVhRm  \

     genre  
3979   pop

In [70]:
df_large_spotify = pd.read_csv(FILE_PATH)

In [71]:
df_large_spotify

id                                 name   
0       35iwgR4jXetI318WEWsa1Q                                Carve  \
1       021ht4sdgPcrDgSk7JTbKY  Capítulo 2.16 - Banquero Anarquista   
2       07A5yehtSnoedViJAZkNnc   Vivo para Quererte - Remasterizado   
3       08FmqUhxtyLTn6pAh6bk45        El Prisionero - Remasterizado   
4       08y9GfoqCWfOGsKdwojr5e                  Lady of the Evening   
...                        ...                                  ...   
586667  5rgu12WBIHQtvej2MdHSH0                                  云与海   
586668  0NuWgxEp51CutD2pJoF4OM                                blind   
586669  27Y1N4Q4U3EfDU5Ubw8ws2            What They'll Say About Us   
586670  45XJsGpFTyzbzeWK8VzR8S                      A Day At A Time   
586671  5Ocn6dZ3BJFPWh4ylwFXtn                     Mar de Emociones   

        popularity  duration_ms  explicit                          artists   
0                6       126903         0                          ['Uli']  \
1                0        98200         0              ['Fernando Pessoa']   
2                0       181640         0              ['Ignacio Corsini']   
3                0       176907         0              ['Ignacio Corsini']   
4                0       163080         0                  ['Dick Haymes']   
...            ...          ...       ...                              ...   
586667          50       258267         0                      ['阿YueYue']   
586668          72       153293         0                   ['ROLE MODEL']   
586669          70       187601         0                      ['FINNEAS']   
586670          58       142003         0  ['Gentle Bones', 'Clara Benin']   
586671          38       214360         0                    ['Afrosound']   

                                               id_artists release_date   
0                              ['45tIt06XoI0Iio4LBEVpls']   1922-02-22  \
1                              ['14jtPCOoNZwquk5wd9DxrY']   1922-06-01   
2                              ['5LiOoJbxVSAMkBS2fUm3X2']   1922-03-21   
3                              ['5LiOoJbxVSAMkBS2fUm3X2']   1922-03-21   
4                              ['3BiJGZsyX9sJchTqcSA7Su']         1922   
...                                                   ...          ...   
586667                         ['1QLBXKM5GCpyQQSVMNZqrZ']   2020-09-26   
586668                         ['1dy5WNgIKQU6ezkpZs4y8z']   2020-10-21   
586669                         ['37M5pPGs6V1fchFJSgCguX']   2020-09-02   
586670  ['4jGPdu95icCKVF31CcFKbS', '5ebPSE9YI5aLeZ1Z2g...   2021-03-05   
586671                         ['0i4Qda0k4nf7jnNHmSNpYv']   2015-07-01   

        danceability  energy  key  loudness  mode  speechiness  acousticness   
0              0.645  0.4450    0   -13.338     1       0.4510         0.674  \
1              0.695  0.2630    0   -22.136     1       0.9570         0.797   
2              0.434  0.1770    1   -21.180     1       0.0512         0.994   
3              0.321  0.0946    7   -27.961     1       0.0504         0.995   
4              0.402  0.1580    3   -16.900     0       0.0390         0.989   
...              ...     ...  ...       ...   ...          ...           ...   
586667         0.560  0.5180    0    -7.471     0       0.0292         0.785   
586668         0.765  0.6630    0    -5.223     1       0.0652         0.141   
586669         0.535  0.3140    7   -12.823     0       0.0408         0.895   
586670         0.696  0.6150   10    -6.212     1       0.0345         0.206   
586671         0.686  0.7230    6    -7.067     1       0.0363         0.105   

        instrumentalness  liveness  valence    tempo  time_signature  
0               0.744000    0.1510   0.1270  104.851               3  
1               0.000000    0.1480   0.6550  102.009               1  
2               0.021800    0.2120   0.4570  130.418               5  
3               0.918000    0.1040   0.3970  169.980               3  
4               0.130000    0.311

In [72]:
df_large_spotify = df_large_spotify.rename(columns={'name' : 'track', 'artists' : 'artist_spotify'})

In [73]:
df_large_spotify['artist_spotify'] = df_large_spotify['artist_spotify'].str.strip('[]').astype(str)
df_large_spotify['artist_spotify'] = df_large_spotify['artist_spotify'].str.strip("'").astype(str)
df_large_spotify


id                                track   
0       35iwgR4jXetI318WEWsa1Q                                Carve  \
1       021ht4sdgPcrDgSk7JTbKY  Capítulo 2.16 - Banquero Anarquista   
2       07A5yehtSnoedViJAZkNnc   Vivo para Quererte - Remasterizado   
3       08FmqUhxtyLTn6pAh6bk45        El Prisionero - Remasterizado   
4       08y9GfoqCWfOGsKdwojr5e                  Lady of the Evening   
...                        ...                                  ...   
586667  5rgu12WBIHQtvej2MdHSH0                                  云与海   
586668  0NuWgxEp51CutD2pJoF4OM                                blind   
586669  27Y1N4Q4U3EfDU5Ubw8ws2            What They'll Say About Us   
586670  45XJsGpFTyzbzeWK8VzR8S                      A Day At A Time   
586671  5Ocn6dZ3BJFPWh4ylwFXtn                     Mar de Emociones   

        popularity  duration_ms  explicit               artist_spotify   
0                6       126903         0                          Uli  \
1                0        98200         0              Fernando Pessoa   
2                0       181640         0              Ignacio Corsini   
3                0       176907         0              Ignacio Corsini   
4                0       163080         0                  Dick Haymes   
...            ...          ...       ...                          ...   
586667          50       258267         0                      阿YueYue   
586668          72       153293         0                   ROLE MODEL   
586669          70       187601         0                      FINNEAS   
586670          58       142003         0  Gentle Bones', 'Clara Benin   
586671          38       214360         0                    Afrosound   

                                               id_artists release_date   
0                              ['45tIt06XoI0Iio4LBEVpls']   1922-02-22  \
1                              ['14jtPCOoNZwquk5wd9DxrY']   1922-06-01   
2                              ['5LiOoJbxVSAMkBS2fUm3X2']   1922-03-21   
3                              ['5LiOoJbxVSAMkBS2fUm3X2']   1922-03-21   
4                              ['3BiJGZsyX9sJchTqcSA7Su']         1922   
...                                                   ...          ...   
586667                         ['1QLBXKM5GCpyQQSVMNZqrZ']   2020-09-26   
586668                         ['1dy5WNgIKQU6ezkpZs4y8z']   2020-10-21   
586669                         ['37M5pPGs6V1fchFJSgCguX']   2020-09-02   
586670  ['4jGPdu95icCKVF31CcFKbS', '5ebPSE9YI5aLeZ1Z2g...   2021-03-05   
586671                         ['0i4Qda0k4nf7jnNHmSNpYv']   2015-07-01   

        danceability  energy  key  loudness  mode  speechiness  acousticness   
0              0.645  0.4450    0   -13.338     1       0.4510         0.674  \
1              0.695  0.2630    0   -22.136     1       0.9570         0.797   
2              0.434  0.1770    1   -21.180     1       0.0512         0.994   
3              0.321  0.0946    7   -27.961     1       0.0504         0.995   
4              0.402  0.1580    3   -16.900     0       0.0390         0.989   
...              ...     ...  ...       ...   ...          ...           ...   
586667         0.560  0.5180    0    -7.471     0       0.0292         0.785   
586668         0.765  0.6630    0    -5.223     1       0.0652         0.141   
586669         0.535  0.3140    7   -12.823     0       0.0408         0.895   
586670         0.696  0.6150   10    -6.212     1       0.0345         0.206   
586671         0.686  0.7230    6    -7.067     1       0.0363         0.105   

        instrumentalness  liveness  valence    tempo  time_signature  
0               0.744000    0.1510   0.1270  104.851               3  
1               0.000000    0.1480   0.6550  102.009               1  
2               0.021800    0.2120   0.4570  130.418               5  
3               0.918000    0.1040   0.3970  169.980               3  
4               0.130000    0.3110   0.1960  103.220               4  
...       

In [74]:
df_large_spotify.duplicated().sum()


0

In [76]:
display(df_large_spotify.isna().sum())
display(df_large_spotify.info())

id                   0
track               71
popularity           0
duration_ms          0
explicit             0
artist_spotify       0
id_artists           0
release_date         0
danceability         0
energy               0
key                  0
loudness             0
mode                 0
speechiness          0
acousticness         0
instrumentalness     0
liveness             0
valence              0
tempo                0
time_signature       0
dtype: int64

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 586672 entries, 0 to 586671
Data columns (total 20 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   id                586672 non-null  object 
 1   track             586601 non-null  object 
 2   popularity        586672 non-null  int64  
 3   duration_ms       586672 non-null  int64  
 4   explicit          586672 non-null  int64  
 5   artist_spotify    586672 non-null  object 
 6   id_artists        586672 non-null  object 
 7   release_date      586672 non-null  object 
 8   danceability      586672 non-null  float64
 9   energy            586672 non-null  float64
 10  key               586672 non-null  int64  
 11  loudness          586672 non-null  float64
 12  mode              586672 non-null  int64  
 13  speechiness       586672 non-null  float64
 14  acousticness      586672 non-null  float64
 15  instrumentalness  586672 non-null  float64
 16  liveness          58

None

In [77]:
df_large_spotify = df_large_spotify.dropna()
df_large_spotify.info()

<class 'pandas.core.frame.DataFrame'>
Index: 586601 entries, 0 to 586671
Data columns (total 20 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   id                586601 non-null  object 
 1   track             586601 non-null  object 
 2   popularity        586601 non-null  int64  
 3   duration_ms       586601 non-null  int64  
 4   explicit          586601 non-null  int64  
 5   artist_spotify    586601 non-null  object 
 6   id_artists        586601 non-null  object 
 7   release_date      586601 non-null  object 
 8   danceability      586601 non-null  float64
 9   energy            586601 non-null  float64
 10  key               586601 non-null  int64  
 11  loudness          586601 non-null  float64
 12  mode              586601 non-null  int64  
 13  speechiness       586601 non-null  float64
 14  acousticness      586601 non-null  float64
 15  instrumentalness  586601 non-null  float64
 16  liveness          586601 

In [80]:
df_large_spotify = df_large_spotify.rename(columns={'artist_spotify' : 'artist'})

In [83]:
#df4 = df_sentiment.merge(df3, on=['track_id','created_at','user_id'], how='inner')
df_spotify_muses = pd.merge(df_large_spotify, df_muse, on=['track'], how='inner' )

In [84]:
df_spotify_muses

id                  track  popularity   
0       35iwgR4jXetI318WEWsa1Q                  Carve           6  \
1       0BRXJHRNGQ3W4v9frnSfhu              Ave Maria           0   
2       0BRXJHRNGQ3W4v9frnSfhu              Ave Maria           0   
3       0BRXJHRNGQ3W4v9frnSfhu              Ave Maria           0   
4       0BRXJHRNGQ3W4v9frnSfhu              Ave Maria           0   
...                        ...                    ...         ...   
164548  2np87QWGVTQeu1zeVxkMDx         Consuming Fire          34   
164549  7a5L9r8Ur8XZGZ98EXVwWs              June Rain          34   
164550  1LDWOI7BzwIZgwxElxuS03           Unfriend You          48   
164551  7xPQY7skgsujvvVyoE5lBi  Beethoven's 5 Secrets          57   
164552  0Zjup4dJep8r3SRZmgrn35             Take Heart          42   

        duration_ms  explicit         artist_x                  id_artists   
0            126903         0              Uli  ['45tIt06XoI0Iio4LBEVpls']  \
1            178933         0      Dick Haymes  ['3BiJGZsyX9sJchTqcSA7Su']   
2            178933         0      Dick Haymes  ['3BiJGZsyX9sJchTqcSA7Su']   
3            178933         0      Dick Haymes  ['3BiJGZsyX9sJchTqcSA7Su']   
4            178933         0      Dick Haymes  ['3BiJGZsyX9sJchTqcSA7Su']   
...             ...       ...              ...                         ...   
164548       477120         0  Hillsong UNITED  ['74cb3MG0x0BOnYNW1uXYnM']   
164549       227960         0            Hu Ge  ['565IvROag6e9RoQoC7a5tj']   
164550       200800         0   Greyson Chance  ['0Qnx1MPnHYt3jJCYrRFVwX']   
164551       309773         0   The Piano Guys  ['0jW6R8CVyVohuUJVcuweDI']   
164552       230413         0  The Sam Willows  ['0mXXLFHmd5bUHxkzaC9ujw']   

       release_date  danceability  energy  ...  time_signature   
0        1922-02-22         0.645   0.445  ...               3  \
1              1922         0.227   0.261  ...               4   
2              1922         0.227   0.261  ...               4   
3              1922         0.227   0.261  ...               4   
4              1922         0.227   0.261  ...               4   
...             ...           ...     ...  ...             ...   
164548   2004-02-01         0.161   0.386  ...               4   
164549   2005-01-21         0.650   0.608  ...               4   
164550   2011-01-01         0.624   0.893  ...               4   
164551   2013-01-18         0.302   0.515  ...               4   
164552   2015-10-30         0.647   0.560  ...               4   

                artist_y                                              seeds   
0           Tara MacLean                               ['wistful', 'light']  \
1                シカタ★アキコ  ['defiant', 'feral', 'negative', 'trashy', 'pr...   
2       Jake Shimabukuro                                         ['gentle']   
3             Nina Hagen                                       ['dramatic']   
4               E Nomine                                      ['enigmatic']   
...                  ...                                                ...   
164548         Third Day                                         ['gloomy']   
164549      Tor Lundvall                                      ['nocturnal']   
164550    Greyson Chance                                         ['bitter']   
164551    The Piano Guys                                       ['powerful']   
164552   Hillsong United                                       ['peaceful']   

        number_of_emotion_tags  valence_tags  arousal_tags  dominance_tags   
0                            5      4.520000      2.832000        4.274000  \
1                           11      4.659460      4.219943        4.426108   
2                            2      6.620000      3.995000        6.625000   
3                            2      4.229211      4.971579        4.237105   
4                            3      1.310000      1.000000        0.795000   
...                        ...           ...        

In [85]:
df_spotify_muses = df_spotify_muses[(df_spotify_muses['artist_x'] == df_spotify_muses['artist_y'])]

In [86]:
df_spotify_muses

id                           track  popularity   
1363    7ALcAwIhNs3hmENPi6wcGI            Till the End of Time          38  \
1513    3npHR2nReF5bhCvvGiNwcL                        Stardust          52   
1569    4YPYFUATiul0877A1HhSlW                        Stardust          55   
1778    6qRvnXftofjYJm1Mg98UWL  Need a Little Sugar in My Bowl          27   
1838    3JRc9ySUX2oxCOUZ3Yr5hB                      Lazy River          23   
...                        ...                             ...         ...   
164511  7j7RgOyPExRMyyp6g83hLh               Jag Skrattar Idag          34   
164516  69KkEYuKWxCjH3taH6z8fz                   Ruina Imperii          48   
164519  6BO5udfgxk73rgdlbwBQh8            Heart Upon My Sleeve          48   
164550  1LDWOI7BzwIZgwxElxuS03                    Unfriend You          48   
164551  7xPQY7skgsujvvVyoE5lBi           Beethoven's 5 Secrets          57   

        duration_ms  explicit          artist_x                  id_artists   
1363         334933         0      Mariah Carey  ['4iHNK0tOyZPYnBU7nGAgpQ']  \
1513         195573         0     Nat King Cole  ['7v4imS0moSyGdXyLgVTIV7']   
1569         194040         0     Nat King Cole  ['7v4imS0moSyGdXyLgVTIV7']   
1778         167640         0      Bessie Smith  ['5ESobCkc6JI4tIMxQttqeg']   
1838         168267         0  Hoagy Carmichael  ['7j8I1aIBA9Z9bMy7mTwWKk']   
...             ...       ...               ...                         ...   
164511       214166         0      Ison & Fille  ['1fhvwCr1HKEZgZYOrfsHyk']   
164516       201133         0           Sabaton  ['3o2dn2O0FCVsWDFSh8qxgG']   
164519       283147         0            Avicii  ['1vCWHaC5f2uS3yhpwWbIA6']   
164550       200800         0    Greyson Chance  ['0Qnx1MPnHYt3jJCYrRFVwX']   
164551       309773         0    The Piano Guys  ['0jW6R8CVyVohuUJVcuweDI']   

       release_date  danceability  energy  ...  time_signature   
1363     1991-09-17         0.475  0.4350  ...               4  \
1513     1957-08-01         0.263  0.3240  ...               3   
1569     1991-01-01         0.188  0.0767  ...               3   
1778           1923         0.693  0.0270  ...               4   
1838           1956         0.644  0.3260  ...               4   
...             ...           ...     ...  ...             ...   
164511   2009-06-09         0.765  0.5800  ...               4   
164516   2012-05-25         0.458  0.7220  ...               4   
164519   2013-01-01         0.707  0.7170  ...               4   
164550   2011-01-01         0.624  0.8930  ...               4   
164551   2013-01-18         0.302  0.5150  ...               4   

                artist_y                     seeds  number_of_emotion_tags   
1363        Mariah Carey                  ['lush']                       8  \
1513       Nat King Cole              ['romantic']                      12   
1569       Nat King Cole              ['romantic']                      12   
1778        Bessie Smith      ['fierce', 'sexual']                       3   
1838    Hoagy Carmichael                  ['lazy']                       1   
...                  ...                       ...                     ...   
164511      Ison & Fille                 ['light']                       2   
164516           Sabaton  ['driving', 'energetic']                       3   
164519            Avicii                  ['epic']                       1   
164550    Greyson Chance                ['bitter']                       1   
164551    The Piano Guys              ['powerful']                       1   

        valence_tags  arousal_tags  dominance_tags   
1363        6.231818      4.204545        5.394545  \
1513        6.952093      3.933953        6.468837   
1569        6.952093      3.933953        6.468837   
1778        3.909552      4.298881        5.670896   
1838        3.050000      3.050000        5.790000   
...              ...           ...             ...   
164511      7.510000      4.785000

In [ ]:
## seulement 9211 entrées communes